src/api/__tests__/Intelligence.test.js

```diff
@@ -144,6 +144,91 @@ describe('api/Intelligence', () => {
+            const suggestionsFromServer = {
+                stringFieldKey: 'fieldVal1',
+                floatFieldKey: 124.0,
+                enumFieldKey: 'EnumOptionKey',
+                multiSelectFieldKey: ['multiSelectOption1', 'multiSelectOption5'],
+            };
+            intelligence.xhr.post = jest.fn().mockReturnValueOnce({
+                data: {
+                    answer: suggestionsFromServer,
+                },
+                create_at: '2025-01-14T00:00:00-00:00',
+                completion_reason: 'done',
+                ai_agent_info: {
+                    processor: 'something',
+                    models: [{ name: 'something', provider: 'something', supported_purpose: 'something' }],
+                },
+            });
+
+            const suggestions = await intelligence.extractStructured(request);
+            expect(suggestions).toEqual(suggestionsFromServer);
+            expect(intelligence.xhr.post).toHaveBeenCalledWith({
+                url: `${intelligence.getBaseApiUrl()}/ai/extract_structured`,
+                id: 'file_123',
+                data: request,
+            });
+        });
+
+        test('should return a succesfull response with correct answer for empty response', async () => {
+            const suggestionsFromServer = {};
+            intelligence.xhr.post = jest.fn().mockReturnValueOnce({
+                data: suggestionsFromServer,
+            });
+            const suggestions = await intelligence.extractStructured(request);
+            expect(suggestions).toEqual(suggestionsFromServer);
+            expect(intelligence.xhr.post).toHaveBeenCalledWith({
+                url: `${intelligence.getBaseApiUrl()}/ai/extract_structured`,
+                id: 'file_123',
+                data: request,
+            });
+        });
+
+        test('should return a succesfull response with correct answer for empty response', async () => {
+            const suggestionsFromServer = {};
+            intelligence.xhr.post = jest.fn().mockReturnValueOnce({
+                data: {
+                    answer: suggestionsFromServer,
+                },
+                create_at: '2025-01-14T00:00:00-00:00',
+                completion_reason: 'done',
+                ai_agent_info: {
+                    processor: 'something',
+                    models: [{ name: 'something', provider: 'something', supported_purpose: 'something' }],
+                },
+            });
+            const suggestions = await intelligence.extractStructured(request);
+            expect(suggestions).toEqual(suggestionsFromServer);
+            expect(intelligence.xhr.post).toHaveBeenCalledWith({
+                url: `${intelligence.getBaseApiUrl()}/ai/extract_structured`,
+                id: 'file_123',
+                data: request,
+            });
+        });
+
+        test.each`
```

Comment:
arent't those test the same as above?

Reply:
Yes, I forgot to remove additional tests after redactor to `test.each`
Thanks for catching this!